File: #179 _Largest_Number/Solution.java
import java.util.*;

public class Solution {

    public String largestNumber(int[] nums) {
        if (nums == null || nums.length == 0) return null;
        String[] strNums = new String[nums.length];
        for (int i = 0; i < nums.length; i++) strNums[i] = Integer.toString(nums[i]);
        StringBuilder sb = new StringBuilder();

        Arrays.sort(strNums, new Comparator<String>() {
        	@Override
        	public int compare(String a, String b) {
        		return -(a + b).compareTo(b + a);
        	}
        });
        for (int i = 0; i < strNums.length; i++) {
        	sb.append(strNums[i]);
        }
        /* delete leading 0's */
        while (sb.charAt(0) == '0' && sb.length() > 1) {
        	sb.deleteCharAt(0);
        }
        return sb.toString();
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	int[] t = {0, 0};
    	System.out.println(sol.largestNumber(t));
    }
}

File: #179 _Largest_Number/Solution.py
class Solution:

    # comparator for string a, b
    def comparator(self, a, b):
        if a + b > b + a:
            return 1
        if a + b < b + a:
            return -1
        return 0

    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        return ''.join(sorted(map(str, nums), self.comparator, reverse=True)).lstrip('0') or '0'


sol = Solution()
print sol.largestNumber(['10'])